lib/model/CDetectionRule.cc

```diff
@@ -54,6 +65,31 @@ bool CDetectionRule::apply(ERuleAction action,
     return true;
 }
 
+void CDetectionRule::executeCallback(CAnomalyDetectorModel& model, core_t::TTime time) const {
+
+    if (m_Callback) {
+        bool executeCallback{true};
+        for (const auto& condition : m_Conditions) {
+            executeCallback &= condition.test(time);
+        }
+        if (executeCallback) {
+            m_Callback(model);
+        }
```

Comment:
Maybe use the efficient early return strategy (although clearly doesn't matter in this case)
```suggestion
        for (const auto& condition : m_Conditions) {
            if (condition.test(time) == false) { // Disjunction
                return;
            }
        }
        m_Callback(model);
```

Reply:
👍🏻